src/app/qgisapp.cpp

```diff
@@ -10127,148 +10128,173 @@ void QgisApp::pasteFromClipboard( QgsMapLayer *destinationLayer )
-        switch ( feedback )
+        connect( dialog, &QgsFixAttributeDialog::finished, this, [ = ]( int feedback )
         {
-          case QgsFixAttributeDialog::PasteValid:
-            //paste valid and fixed, vanish unfixed
-            features << validFeatures << dialog->fixedFeatures();
-            break;
-          case QgsFixAttributeDialog::PasteAll:
-            //paste all, even unfixed
-            features << validFeatures << dialog->fixedFeatures() << dialog->unfixedFeatures();
-            break;
-        }
-        pasteFeatures( pasteVectorLayer, invalidGeometriesCount, nTotalFeatures, features );
-        dialog->deleteLater();
-      } );
-      dialog->show();
-      return;
+          QgsFeatureList features = pastedFeatures;
+          switch ( feedback )
+          {
+            case QgsFixAttributeDialog::PasteValid:
+              //paste valid and fixed, vanish unfixed
+              features << validFeatures << dialog->fixedFeatures();
+              break;
+            case QgsFixAttributeDialog::PasteAll:
+              //paste all, even unfixed
+              features << validFeatures << dialog->fixedFeatures() << dialog->unfixedFeatures();
+              break;
+          }
+          pasteFeatures( pasteVectorLayer, invalidGeometriesCount, nTotalFeatures, features );
+          dialog->deleteLater();
+        } );
+        dialog->show();
+        return;
+      }
     }
   }
 
-  pasteFeatures( pasteVectorLayer, invalidGeometriesCount, nTotalFeatures, newFeatures );
+  pasteFeatures( pasteVectorLayer, invalidGeometriesCount, nTotalFeatures, pastedFeatures, duplicateFeature );
 }
 
-void QgisApp::pasteFeatures( QgsVectorLayer *pasteVectorLayer, int invalidGeometriesCount, int nTotalFeatures, QgsFeatureList &features )
+void QgisApp::pasteFeatures( QgsVectorLayer *pasteVectorLayer, int invalidGeometriesCount, int nTotalFeatures, QgsFeatureList &features, bool duplicateFeature )
 {
   int nCopiedFeatures = features.count();
-  if ( pasteVectorLayer->addFeatures( features ) )
+  QString childrenInfo;
+  QgsFeatureIds newIds;
+  newIds.reserve( features.size() );
+  if ( duplicateFeature )
   {
-    QgsFeatureIds newIds;
-    newIds.reserve( features.size() );
     for ( const QgsFeature &f : std::as_const( features ) )
     {
-      newIds << f.id();
-    }
+      QgsVectorLayerUtils::QgsDuplicateFeatureContext duplicateFeatureContext;
+      QgsFeature duplicatedFeature = QgsVectorLayerUtils::duplicateFeature( pasteVectorLayer, f, QgsProject::instance(), duplicateFeatureContext );
+      newIds << duplicatedFeature.id();
 
-    pasteVectorLayer->selectByIds( newIds );
+      const auto duplicateFeatureContextLayers = duplicateFeatureContext.layers();
+      for ( QgsVectorLayer *chl : duplicateFeatureContextLayers )
+      {
+        childrenInfo += ( tr( "\n%n children on layer %1 duplicated", nullptr, duplicateFeatureContext.duplicatedFeatures( chl ).size() ).arg( chl->name() ) );
```

Comment:
Won't this resort in one "xxx children on layer" message PER feature pasted? That's potentially 100s/1000s ....